src/devtools/client/webconsole/components/FilterBar/FilterBar.js

```diff
@@ -136,11 +140,22 @@ class FilterBar extends Component {
   }
 
   renderClearButton() {
-    const { messagesClearEvaluations } = this.props;
+    const { messagesClearEvaluations, allMessagesById } = this.props;
+    let messagesExist = false;
+
+    // Get all messages related to evaluations and determine whether or not we
+    // have any, if we don't we want to disable the trash button
+    for (const [id, message] of allMessagesById) {
+      if (message.type === MESSAGE_TYPE.COMMAND || message.type === MESSAGE_TYPE.RESULT) {
+        messagesExist = true;
+        break;
+      }
+    }
 
     return dom.button({
       className: "devtools-button devtools-clear-icon",
-      title: "Clear the Web Console output",
+      title: messagesExist ? "Clear the Web Console output" : "No outputs to clear",
```

Comment:
I'd make this more descriptive, like "Clear console evaluations" and "No console evaluations to clear"

Reply:
fixed